iterativeTrimParallel.py

```diff
@@ -0,0 +1,593 @@
+#!/bin/env python
+"""
+Script to set trimdac values on a V3 chamber iteratively
+By: Brian Dorney (brian.l.dorney@cern.ch)
+
+Based on algorithmic description: https://indico.cern.ch/event/838248/contributions/3515801/attachments/1887448/3112771/VFAT3b_trim.pdf
+"""
+
+import os
+import ROOT as r
+
+from gempython.utils.gemlogger import printGreen, printYellow, printRed
+from gempython.vfatqc.utils.qcutilities import getCardName
+from gempython.tools.amc_user_functions_xhal import HwAMC
+from gempython.gemplotting.mapping.chamberInfo import chamber_config, GEBtype
+
+def iterativeTrim(args,dict_dirPaths,identifier,dict_chanRegData=None,dict_calFiles=None):
+    """
+    Takes an scurve at a given set of channel registers (all 0's if not provided) and
+    returns a dictionary of numpy arrays with fit results (see gempythong.gemplotting.fitting.fitScanData
+    for details on the output container).
+
+    args        - Namespace produced by ArgumentParser.parse_args
+    dirPath     - Output filepath location that scurve raw data should be saved at
+    identifier  - Unique string identifier to be used for each iteration
+    chanRegData - structure numpy array containing channel register data, expected dtype values are:
+                  [('CALPULSE_ENABLE','bool'),('MASK','bool'),('ZCC_TRIM_POLARITY','bool'),
+                  ('ZCC_TRIM_AMPLITUDE','uint8'),('ARM_TRIM_POLARITY','bool'),('ARM_TRIM_AMPLITUDE','uint8')].
+                  If provided these values will be written to the channel registers of all unmaksed VFATs.
+    calInfo     - Tuple of numpy arrays providing CFG_CAL_DAC calibration info, idx = 0 (1) for slopw (intercept)
+                  indexed by VFAT position
+    """
+
+    from ctypes import c_uint32
+
+    import root_numpy as rp
+
+    from gempython.vfatqc.utils.scanUtils import launchSCurve
+
+    dictOfFiles = {}
+
+    for ohN in range(0,amcBoard.nOHs+1):
+        # Skip masked OH's
+        if( not ((args.ohMask >> ohN) & 0x1)):
+            continue    
+
+        # Get Channel Register Info
+        if dict_chanRegData is None:
+            setChanRegs = False
+            cArray_trimVal = None
+            cArray_trimPol = None
+        else:
+            setChanRegs = True
+            cArray_trimVal = (c_uint32 * 3072)(*dict_chanRegData[ohN]["ARM_TRIM_AMPLITUDE"])
```

Comment:
the 3072 constant should not be hardcoded, but deduced from them gem type (for GE1/1 it is 3072, for GE2/1 it is 1536, for ME0 - tbd)

Reply:
updated with `vfatsPerGemVariant`